File: 345. Reverse Vowels of a String.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowel=[]
        check=['a','e','i','o','u','A','E','I','O','U']
        for i in s:
            if i in check:
                vowel.append(i)
        
        res=""
        
        for i in s:
            if i in check and len(vowel)>0:
                # print(res)
                res+=vowel.pop()
                continue
            res+=i
        return res

File: 345. Reverse Vowels of a String.java
class Solution {
    public String reverseVowels(String s) {
        List<Character> vowel = new ArrayList<>();
        char[] check = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        
        for (char i : s.toCharArray()) {
            for (char c : check) {
                if (i == c) {
                    vowel.add(i);
                    break;
                }
            }
        }
        
        StringBuilder res = new StringBuilder();
        
        for (char i : s.toCharArray()) {
            boolean isVowel = false;
            for (char c : check) {
                if (i == c) {
                    isVowel = true;
                    break;
                }
            }
            if (isVowel && vowel.size() > 0) {
                res.append(vowel.remove(vowel.size() - 1));
                continue;
            }
            res.append(i);
        }
        return res.toString();
    }
}
